### ISSUE TITLE
[Suggestion] Prioritize egg hatching

### ISSUE THREAD

**Issue by progz** [NECROBOTIO/NecroBot#3233]
Hi,
Should be great if we can chose what kind of eggs I would prefer to hatch first.

Exemple of configuration 

> "EggToHatchInPriority": [
>     { "Key": "2", "Value": 100 },
>     { "Key": "5", "Value": 50 },
>     { "Key": "10", "Value": 75 }
> ]

The biggest value will be hatch in first, then the second value and to finish the lowest value.
In this example, it will hatch in first 2 Km egg then 10 km then 5 km.

**Comment by Simpuhl** [comment]
I believe it already does this. (Hatches 10K > 5k > 2k) in that order. Mine always does 10k if I have 10k and then 5k if I have no 5k, and 2k if I have no 10k or 5k.

But your asking to customize it?

**Comment by progz** [comment]
yeah for exemple I really need to prioritize 2Km eggs and I almost never hatch it with the bot.

**Comment by Simpuhl** [comment]
Why does it matter tho?

If you have a 10k or 5, might as well do those... the issue is as soon as u hatch a 2k ur probably just going to find another 2k... lol

**Comment by progz** [comment]
Maybe. 
I just think it should be great to let decide. For me, the time I hatch my 10 and 5, I will find another some 10 some 5 and of course some 2. But the bot will hatch in priority  the 10 again...

And you know we can also imagine a parameter as : 
_PrioritizeEggHatching : "False|true|random"_

- False does the current system
- True respect the list "EggToHatchInPriority"
- Random does a random pick up

**Comment by progz** [comment]
If you need an explanation, just ask :)

**Comment by schwartzadev** [comment]
Interesting. I could see the application esp if you want to hatch a Pokemon that hatches from a 2km egg

**Comment by Flerio** [comment]
i pray for this change xD
